tests/test_clean.py

```diff
@@ -1,96 +1,210 @@
+import os
+
 from html5lib.filters.base import Filter
 import pytest
 
-import bleach
+from bleach import clean
 from bleach.sanitizer import Cleaner
 
 
-def test_empty():
-    assert bleach.clean('') == ''
+def test_clean_idempotent():
+    """Make sure that applying the filter twice doesn't change anything."""
+    dirty = '<span>invalid & </span> < extra http://link.com<em>'
+    assert clean(clean(dirty)) == clean(dirty)
 
 
-def test_nbsp():
-    assert bleach.clean('&nbsp;test string&nbsp;') == '&nbsp;test string&nbsp;'
+def test_only_text_is_cleaned():
+    some_text = 'text'
+    some_type = int
+    no_type = None
 
+    assert clean(some_text) == some_text
 
-def test_comments_only():
-    comment = '<!-- this is a comment -->'
-    assert bleach.clean(comment) == ''
-    assert bleach.clean(comment, strip_comments=False) == comment
+    with pytest.raises(TypeError) as e:
+        clean(some_type)
+    assert "argument cannot be of 'type' type" in str(e)
 
-    open_comment = '<!-- this is an open comment'
-    assert bleach.clean(open_comment) == ''
-    assert (
-        bleach.clean(open_comment, strip_comments=False) ==
-        '{0!s}-->'.format(open_comment)
-    )
+    with pytest.raises(TypeError) as e:
+        clean(no_type)
+    assert "NoneType" in str(e)
```

Comment:
This is the same test--I moved it to the top figuring "this is testing a property of `clean`".